articles/node-url-shortener/index.md

```diff
@@ -0,0 +1,231 @@
+Using links or URLs has been the norm of surfing the web for a long time. Sometimes we need to advertise our businesses on social media. Long URL links are not the best way to post links, especially on social media. This will help you to promote your products or any services with the links you are providing. The benefit of URL shortening tools is helping in driving the traffic back to your website. In this tutorial, we will build a URL shortener service using Node.js.
+
+### Prerequisites
+1. A basic understanding of the [JavaScript](https://www.w3schools.com/js/DEFAULT.asp) programming language.
+   
+2. A basic understanding of [Node.js](https://nodejs.org/en/) and [Express](https://expressjs.com/) framework.
+   
+3. Have [Postman](https://www.postman.com/downloads/) HTTP API client installed on your system.
+   
+4. A text editor, preferably [VS Code](https://code.visualstudio.com/download).
+
+5. [MongoDB](https://www.mongodb.com/try/download/community) database server installed on your system.
+
+### Setting Up The Project
+Create a folder named `URL-Shortener-Service` and open it on your favorite IDE. I will be using VS Code here. Go inside the folder and type `npm init` to generate an initial package for our project. This command will require some prompts, to skip this, you can use `npm init -y`. Our project will use various npm packages from the npm registry. The npm packages include:
+
+- `express`: Express is a backend web application framework for Node.js for building web applications and APIs.
+
+- `mongoose`: Mongoose is an asynchronous database driver or Object Data Mapper for MongoDB. It is responsible for connecting to the database and performing query operations.
+
+- `short-id`: The `short-id` module creates user-friendly and unique ids for our URLs.
+
+- `valid-url`: This is a module that verifies all the URLs sent to the API.
+
+- `nodemon`: nodemon package will be installed as a development dependency. It will constantly monitors our application by automatically restarting the server when any file changes.
+  
+Next, we now need to download the packages. Run the following command on your terminal to download them:
+
+```bash
+npm install express mongoose shortid valid-url
+```
+
+This will download the named packages inside `node_modules` folder and update the `package.json` file with the dependencies.
+
+>>> Note: `nodemon` is a development dependency. To install this, run the command `npm install --save-dev nodemon`.
+
+
+### Setting Up Express server
+Inside our `URL-Shortener-Service` folder, create a file named `server.js`. The following is the initial code to start the Express server:
+
+```js
```

Comment:
Please add code comments here. 
I know that you've explained the code, but comments make the code more readable.